KStudy.js

```diff
@@ -0,0 +1,509 @@
+					},
+					{
+						"tag": "은어"
+					},
+					{
+						"tag": "청소년 언어"
+					}
+				],
+				"notes": [],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "http://kiss.kstudy.com/journal/journal-view.asp?key1=25169&key2=2201",
+		"items": "multiple"
+	},
+	{
+		"type": "web",
+		"url": "http://kiss.kstudy.com/public/public2-article.asp?key=50064290",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "치과용 콘빔 CT를 이용한 상악 정중과잉치의 3차원 분석",
+				"creators": [
+					{
+						"lastName": "이동호",
+						"creatorType": "author",
+						"fieldMode": 1
+					},
+					{
+						"lastName": "이재서",
+						"creatorType": "author",
+						"fieldMode": 1
+					},
+					{
+						"lastName": "윤숙자",
+						"creatorType": "author",
+						"fieldMode": 1
+					},
+					{
+						"lastName": "강병철",
+						"creatorType": "author",
+						"fieldMode": 1
+					}
+				],
+				"date": "2010",
+				"ISSN": "1229-8212",
+				"abstractNote": "Purpose : This study was performed to analyze the position, pattern of impacted mesiodens, and their relationship to the adjacent teeth using Dental cone-beam CT. Materials and Methods : Sixty-two dental cone-beam CT images with 81 impacted mesiodenses were selected from about 2,298 cone-beam CT images at Chonnam National University Dental Hospital from June 2006 to March 2009. The position, pattern, shape of impacted mesiodenses and their complications were analyzed in cone-beam CT including 3D images. Results : The sex ratio (M : F) was 2.9 : 1. Most of the mesiodenses (87.7%) were located at palatal side to the incisors. 79% of the mesiodenses were conical in shape. 60.5% of the mesiodenses were inverted, 21% normal erupting direction, and 18.5% transverse direction. The complications due to the presence of mesiodenses were none in 43.5%, diastema in 19.4%, tooth displacement in 17.7%, delayed eruption or impaction in 12.9%, tooth rotation in 4.8%, and dentigerous cyst in 1.7%. Conclusions : Dental cone-beam CT images with 3D provided 3-dimensional perception of mesiodens to the neighboring teeth. This results would be helpful for management of the impacted mesiodens.",
+				"issue": "3",
+				"journalAbbreviation": "대한구강악안면방사선학회지 (대한구강악안면방사선학회)",
+				"language": "ko-KR",
+				"libraryCatalog": "KStudy",
+				"pages": "109–114",
+				"publicationTitle": "대한구강악안면방사선학회지 (대한구강악안면방사선학회)",
+				"url": "http://kiss.kstudy.com/public/public2-article.asp?key=50064290",
+				"volume": "40",
+				"attachments": [],
+				"tags": [
+					{
+						"tag": "Cone-Beam Computed Tomograpahy"
+					},
+					{
+						"tag": "Incisor"
+					},
+					{
+						"tag": "Supernumerary"
+					},
+					{
+						"tag": "Tooth"
+					}
+				],
+				"notes": [],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "http://kiss.kstudy.com/public/public2-article.asp?key=50789039",
+		"items": [
+			{
+				"itemType": "journalArticle",
+				"title": "KOLMOGOROV DISTANCE FOR MULTIVARIATE NORMAL APPROXIMATION",
```

Comment:
let's title case all caps titles

Reply:
I tried to think about this before. I don't know how to check easily whether the title is all-uppercase but not a Korean title. Any ideas?

Reply:
You should be able to just use `ZU.capitalizeTitle(title, true)`.

Reply:
Thank you. This also changes the title `치과용 콘빔 CT를 이용한 상악 정중과잉치의 3차원 분석`, which I wanted to avoid. I guess some test like `[\u0000-\u00FF]*` should work for Latin alphabet only titles...

Reply:
How does it change it?

Reply:
So it lowercases the 't'. I haven't looked at this function too closely, but that's plausibly just a bug that we should fix.

Reply:
Yes, it makes `Ct`.

But I prefer to use `capitalizeTitle` only when we have all-uppercase titles. Otherwise also names like `USA` or `EU` will be strangely capitalize. See also the used pattern in other translators like in https://github.com/zotero/translators/blob/e04633baeaac2c62eddb4435e517ae7d51c89a73/Web%20of%20Science.js#L131-L133

Reply:
`capitalizeTitle` won't change "USA" and "EU" unless the string is fully capitalized. But it will convert things that are in sentence case to title case, which we don't want to do unnecessarily (if the capitalizeTitles pref is left off).

```
>> Zotero.Utilities.capitalizeTitle("The USA and also the EU", true)
"The USA and Also the EU"
```

```
>> Zotero.Utilities.capitalizeTitle("THE USA AND ALSO THE EU", true)
"The Usa and Also the Eu"
```

If we want to avoid messing up things like the `Ct`, we could perhaps say that `capitalizeTitle` shouldn't change words if they contain any characters outside `[\u0000-\u00FF]`. I'm not positive that's always what we want, though.

Reply:
it will be great if the English title can be normalized, taking capitalization e.g. proper noun into account, and other words such as USA, EU in capitalize form. There are some similar projects (https://github.com/nlp-compromise/compromise), but I haven't found a "killer Javascript function" yet.

Reply:
If a site gives a title in all-caps, as seems to be the case here, we generally just title case it according to some basic rules and leave the fixing of proper nouns and acronyms up to users. (Along other things, we don’t even necessarily know the language in order to reliably exclude words.)

The issue here just seems to be (I haven’t looked at this closely) that some Korean titles include English characters, and we want to avoid changing the case of those, either in the translator or in a more generalized way in `capitalizeTitle`.

Reply:
@dstillman Okay, my mistake. But the code pattern can still make sense then, to prevent making title case, when sentence case is correctly given.

I test that the majority of the title is from Latin alphabet. But your suggestion for `capitalizeTitle` on the words

> not change words if they contain any characters outside [\u0000-\u00FF]

sounds reasonable for me.

@socheres The function `ZU.capitalizeTitle` can be used to "normalize" English titles into title case  in translators, see https://github.com/zotero/zotero/blob/master/chrome/content/zotero/xpcom/utilities.js#L921 . However, it will add a capital letter to more or less every word with only a small number of exceptions cf. https://github.com/zotero/zotero/blob/2baa537542fcd5bc6e2826eb0eef6b34c571dcfc/chrome/content/zotero/xpcom/utilities.js#L922-L924 . To do anything similar e.g. for German and thereby differentiating between nouns and verbs etc. might be out of scope for Zotero.

Reply:
thank you @avram @zuphilip for explaining. i totally understand that fixing crappy data records is out of scope for zotero. it was just feedback - nice to have feature - from the users/scientists. publisher who create metadata have to deal it.